docs/modules/ROOT/pages/BasicConcept.adoc

```diff
@@ -36,56 +53,180 @@ filter4 = "Power Nice" -- nick filter maches noone
+
+*Functions*
+
+A *function* is a capability that is associated with a class. For example, the class `Manufacturer` has a function called `setRecipe` that, when given the name of a recipe, actually changes the active recipe on a Manufacturer!
+
+_Note: If a lot of these descriptions seem obvious (e.g. "setRecipe sets a recipe"), that's because the system itself is designed to make sense to a human reading it--albeit in a highly technical manner. The more you use it, the better you'll get at reading the documentation and understanding its functions intuitively._
+
+==== Class Inheritance
+
+Earlier we mentioned that every object in the game is made up of a blueprint called a class. In fact, each object is constructed of _multiple_ classes. This is because of a principal called *class inheritance*, which allows an object to inherit the functions and properties of _each_ class that constructs it.
+
+For example, a look at the documentation shows a class called `Actor` (also called `FGBuildable`) with the description, "This is the base class of all things that can exist within the world by themselves." From that description, we can infer that it contains properties (e.g. `location`, `rotation`) and functions (e.g. `getInventories`) that are common to _all_ buildables. Because we've expressed these characteristics in a *parent class*, everything that falls under that class (known as *child classes*) inherits them without having to repeatedly include them in their description!
+
+That's useful when for _building_ a system, but what does it mean when you're _using_ it? In short, when referring to the Reflection docs, you should understand that any class you reference has the functions and properties of itself and _every_ parent it has (found by clicking link labeled "Parent" at the top of its description). Doing so will help make the reference easier to understand and more powerful to use.
+
+=== Types
+
+For code to run properly, functions must receive properly formatted inputs and return properly formatted outputs. In order to ensure that this happens every single time, the Reflection System uses types. 
+
+*Types* in coding are simply definitions that prescribe what kind of value something can have. When a value is entered into a function that does not match its pre-defined input type, the code fails in order to prevent it from doing unexpected things.
+
+==== Simple Values
+
+Individual values are expressed using basic types. Here are the most common:
+
+- `String` - Sequence of letters and other characters
+
+- `Int` - Integer, e.g. 1, 35, 7009
+
+- `Float` - A number with a floating decimal point, e.g. 1.23456
+
+- `bool` - 'true' or 'false'
+
+==== Complex Types
+
+Collections of values use different types based on the types of the items in them.
+
+- `Array` - Ordered collection of data of the _same_ type
+
+- `Struct` - Unordered collection of data that can contain _different_ types
+
+- `Class` - Unordered collection of data and functions
```

Comment:
Class is not a Complex type like Array...
Its more somthing like Trace or Object...
The key difference is, this Class refers to a "Class Instance", its something that is rather unique to how Satisfactory handles some stuff...
You could view it as a modular Enum.
F.e. Item-Types are implemented as seperated classes, one class for each type of item. All of them exted UFGItem.
When working with them internally, you dont create a instance of those classes, instead we pass the references to the Unreal Reflection Objects of those classes, and then use static functions and the CDO to retrieve information from them abstractly (like the name).
To make it more easy to use, I created this Class Instance system, that allows to access this data without the need of any static functions etc.